pkg/logging/logger.go

```diff
@@ -102,8 +124,17 @@ func (p NamedLogProducer) Error(message string, err interface{}) {
 	p.log(LogLevelError, message)
 }
 
+var defaultLogConsumer = NewFilteredLevelLogConsumer(LogLevelInfo, os.Stdout)
+var loggerLock = &sync.Mutex{}
+
 func (p NamedLogProducer) log(logLevel LogLevel, message string) {
-	mutex.Lock()
-	defaultLogConsumer.Log(logLevel, message, p.fields)
-	mutex.Unlock()
+	if logger := getLogger(GetSdkKey(p.ctx)); logger != nil {
+		loggerLock.Lock()
+		logger.Log(logLevel, message, p.fields)
+		loggerLock.Unlock()
+	} else {
+		loggerLock.Lock()
+		defaultLogConsumer.Log(logLevel, message, p.fields)
+		loggerLock.Unlock()
+	}
```

Comment:
Per the pervious comment about using `fields` to propagate the SDK key, I don't think we need to support multiple SDK consumers anymore.